tests/unit/modules/network/vyos/test_vyos_bgp_global14.py

```diff
@@ -0,0 +1,567 @@
+                            maximum_prefix=34,
+                            update_source="2001:db8::1",
+                        ),
+                    ],
+                    network=[
+                        dict(address="172.16.42.32/27", backdoor=True),
+                        dict(address="172.16.42.251/32", route_map="map01"),
+                    ],
+                    bgp_params=dict(
+                        bestpath=dict(as_path="confed", compare_routerid=True),
+                        default=dict(no_ipv4_unicast=True),
+                        router_id="10.1.1.1",
+                    ),
+                    redistribute=[
+                        dict(protocol="kernel", route_map="map01"),
+                        dict(protocol="static", metric=20),
+                        dict(protocol="static", route_map="map01"),
+                    ],
+                ),
+                state="rendered",
+            ),
+        )
+        rendered_cmds = [
+            "set protocols bgp system-as 65536",
+            "set protocols bgp neighbor 10.0.0.4 disable-connected-check",
+            "set protocols bgp neighbor 10.0.0.4 timers holdtime 30",
+            "set protocols bgp neighbor 10.0.0.4 timers keepalive 10",
+            "set protocols bgp neighbor 10.0.0.4 capability orf prefix-list receive",
+            "set protocols bgp neighbor 192.168.0.2 attribute-unchanged as-path",
+            "set protocols bgp neighbor 192.168.0.2 attribute-unchanged med",
+            "set protocols bgp neighbor 192.168.0.2 attribute-unchanged next-hop",
+            "set protocols bgp neighbor 192.168.0.2 ebgp-multihop 2",
+            "set protocols bgp neighbor 192.168.0.2 remote-as 65535",
+            "set protocols bgp neighbor 192.168.0.2 soft-reconfiguration",
+            "set protocols bgp neighbor 192.168.0.2 update-source 192.168.0.1",
+            "set protocols bgp neighbor 2001:db8::2 ebgp-multihop 2",
+            "set protocols bgp neighbor 2001:db8::2 remote-as 65535",
+            "set protocols bgp neighbor 2001:db8::2 maximum-prefix 34",
+            "set protocols bgp neighbor 2001:db8::2 update-source 2001:db8::1",
+            "set protocols bgp redistribute kernel route-map map01",
+            "set protocols bgp redistribute static route-map map01",
+            "set protocols bgp network 172.16.42.32/27 backdoor",
+            "set protocols bgp network 172.16.42.251/32 route-map map01",
+            "set protocols bgp parameters bestpath as-path confed",
+            "set protocols bgp parameters bestpath compare-routerid",
+            "set protocols bgp parameters default no-ipv4-unicast",
+            "set protocols bgp parameters router-id 10.1.1.1",
+        ]
+        result = self.execute_module(changed=False)
+        self.assertEqual(
+            sorted(result["rendered"]),
+            sorted(rendered_cmds),
+            result["rendered"],
+        )
+
+    def test_vyos_bgp_global_parsed(self):
```

Comment:
If you haven't started looking at the integration tests, one that will be interesting is the `parsed` integration tests, because there may be a need to know which version you're using before parsing the data. In this case, we need to consider the correct default (i.e. if the user doesn't force the vyos version through the config variables) and whether it should be 1.2, 1.3, or 1.4. Since that may affect the parsing.

Alternately, we could look at making the default parser resilient against the missing or extra parameter if need be.

This can be tested locally with a simple ansible file containing similar commands to below and the ones in the corresponding "old" config to see how they are parsed without any manual setting of the os version, and then with it.

Reply:
The integrity tests were OK now, thanks @gaige for help!